lib/go-parser/index.ts

```diff
@@ -0,0 +1,114 @@
+import * as elf from "elfy";
+import * as path from "path";
+import { Readable } from "stream";
+import { ExtractAction } from "../extractor/types";
+
+const ignoredPaths = [
+  path.normalize("/boot"),
+  path.normalize("/dev"),
+  path.normalize("/etc"),
+  path.normalize("/home"),
+  path.normalize("/media"),
+  path.normalize("/mnt"),
+  path.normalize("/proc"),
+  path.normalize("/root"),
+  path.normalize("/run"),
+  path.normalize("/sbin"),
+  path.normalize("/sys"),
+  path.normalize("/tmp"),
+  path.normalize("/var"),
+];
+
+function filePathMatches(filePath: string): boolean {
+  const dirName = path.dirname(filePath);
+  return (
+    !path.parse(filePath).ext &&
+    !ignoredPaths.some((ignorePath) => dirName.startsWith(ignorePath))
+  );
+}
+
+export const getGoModulesContentAction: ExtractAction = {
+  actionName: "gomodules",
+  filePathMatches,
+  callback: findGoBinaries,
+};
+
+async function findGoBinaries(
+  stream: Readable,
+  streamSize?: number,
+): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const encoding = "binary";
+    const buildIdMagic = "Go";
+    const elfHeaderMagic = "\x7FELF";
+    const buildInfoMagic = "\xff Go buildinf:";
+
+    const buffer: Buffer = Buffer.alloc(streamSize ?? 64 * 1024);
```

Comment:
I think 64KB might be tiny, I would be happy to pre-allocate even something as big as 4MB or 8MB 😄